test/integrationtests.jl

```diff
@@ -769,4 +769,71 @@ end
     @test ts.time_end - ts.time_start ≈ timeout
 end
 
+@testset "worker always crashes immediately" begin
+    file = joinpath(TEST_FILES_DIR, "_happy_tests.jl")
+
+    # We have occassionally seen the Process exist with the expected signal.
+    @assert typemin(Int32) == -2147483648
+    terminated_err_log_1 = r"Error: Worker\(pid=\d+, terminated=true, termsignal=(6|-2147483648)\) terminated unexpectedly. Starting new worker \(retry 1/2\)."
+    terminated_err_log_2 = r"Error: Worker\(pid=\d+, terminated=true, termsignal=(6|-2147483648)\) terminated unexpectedly. Starting new worker \(retry 2/2\)."
+
+    worker_init_expr = :(@eval ccall(:abort, Cvoid, ()))
+    # We don't use IOCapture for capturing logs as that seems to hang when the worker crashes.
+    mktemp() do io, path
+        results = redirect_stdio(stdout=io, stderr=io, stdin=devnull) do
+            encased_testset() do
+                runtests(file; nworkers=1, worker_init_expr)
+            end
+        end
+        captured = read(path, String)
+        # Test we retried starting a worker twice and saw the expected log each time.
+        @test contains(captured, terminated_err_log_1)
+        @test contains(captured, terminated_err_log_2)
+        # Test that `runtests` errored overall, before any test items were run.
+        @test n_tests(results) == 1
+        @test length(errors(results)) == 1
+    end
+end
+
+@testset "worker crashes immediately but succeeds on retry" begin
+    file = joinpath(TEST_FILES_DIR, "_happy_tests.jl")
+    nworkers = 4
+    tmpdir = joinpath("/tmp", "JL_RETESTITEMS_TEST_TMPDIR")
```

Comment:
there was a reason we couldn't do that ...although i can't remember what it was

Reply:
added a comment

Reply:
Hmm... The `cleanup` functionality should be tied to the calling process (the one that calls `runtests`), not the spawned worker processes, so that should be ok, no? We can also do manual cleanup with `mktempdir(...;cleanup=false)`. Also, `/tmp` would fail on windows, `tmpdir()`  should handle this across platforms.

Reply:
i remember we previously had issues with `tmpdir` so i'm reluctant to go back

i don't think we need be concerned about the tests for ReTestItems running on windows

Reply:
ah, i see how the other case of `/tmp` is different -- in this PR i think you're right and `mktemp` is fine

Reply:
Thanks! Yeah I think if we have some contributors using Windows, this would make it a bit easier for them.